robot.py

```diff
@@ -130,19 +130,46 @@ def handle_hang_inputs(self, joystick: wpilib.Joystick):
             self.hang.winch()
 
     def testInit(self):
-        self.turret.motor.configPeakOutputForward(0.15, 10)
-        self.turret.motor.configPeakOutputReverse(-0.15, 10)
+        self.turret.motor.configPeakOutputForward(1.0, 10)
+        self.turret.motor.configPeakOutputReverse(-1.0, 10)
+        self.test_hall_effect_found = False
 
     def testPeriodic(self):
         # Slew the turret
-        five_degrees = math.radians(5)  # radians
-        if self.driver_joystick.getRawButtonPressed(5):
-            self.turret.slew(five_degrees)
+        slew_increment = math.radians(5)  # radians
+        # TODO turret angles are reversed
+        if self.driver_joystick.getRawButtonPressed(6):
+            self.turret.slew(slew_increment)
+            self.turret.execute()
+        elif self.driver_joystick.getRawButtonPressed(5):
+            self.turret.slew(-slew_increment)
             self.turret.execute()
-        elif self.driver_joystick.getRawButtonPressed(6):
-            self.turret.slew(-five_degrees)
+        if self.driver_joystick.getRawButtonPressed(2):
+            self.turret.scan()
+        if self.turret.current_state == self.turret.SCANNING:
             self.turret.execute()
 
+        # Find the width of a hall effect sensor
+        if self.driver_joystick.getRawButtonPressed(12):
+            self.turret.motor.set(ctre.ControlMode.PercentOutput, 0.25)
+        if (
+            self.turret.centre_index.get() == self.turret.HALL_EFFECT_CLOSED
+            and not self.test_hall_effect_found
+        ):
+            print(
```

Comment:
Print calls can be expensive, and when an RPC is involved, of indeterminate latency. It might help reduce the uncertainty of this measurement if we capture the two positions, then print both of them afterwards.